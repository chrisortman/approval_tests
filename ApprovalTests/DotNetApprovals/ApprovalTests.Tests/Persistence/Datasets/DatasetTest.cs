using System;
using System.Data;
using System.Reflection;
using ApprovalTests.Persistence.DataSets;
using NUnit.Framework;
using ReportingDemo;

namespace ApprovalTests.Tests.Persistence.Datasets
{
	[TestFixture]
	public class DatasetTest
	{
		private const string ReportName = "ReportingDemo.InsultsReport.rdlc";

		private static DataTable GetDefaultData()
		{
			return new InsultsDataSet.InsultsDataTable().AddTestDataRows();
		}
		private static Assembly GetAssembly()
		{
			return typeof (InsultsDataSet).Assembly;
		}

		[Test]
		public void TestSimpleReportWith1Dataset()
		{
			RdlcReports.Approvals.ApproveReport(ReportName, GetDefaultData());
		}

		[Test]
		public void TestSimpleReportWithDatasetInAssembly()
		{
			RdlcReports.Approvals.ApproveReport(ReportName, "Model", GetDefaultData());
		}

		[Test]
		public void TestReport()
		{
			RdlcReports.Approvals.ApproveReport(ReportName, GetAssembly(), Tuple.Create("Model", GetDefaultData()));
		}


		[Test]
		public void TestSimpleReportExplict()
		{
			RdlcReports.Approvals.ApproveReport(ReportName, GetAssembly(), "Model", GetDefaultData());
		}

		[Test]
		public void TestDataSourceNames()
		{
			var message = "";
			try
			{
				RdlcReports.Approvals.ApproveReport(ReportName, GetAssembly(), "purposelyMisspelt", GetDefaultData());
			}
			catch (Exception e)
			{
				message = e.Message;
			}
			Approvals.Approve(message);
		}
	}
}