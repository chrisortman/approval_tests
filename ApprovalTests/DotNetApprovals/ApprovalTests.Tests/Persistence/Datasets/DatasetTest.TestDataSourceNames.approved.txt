The Datasource Name 'purposelyMisspelt'
is not a legal match for ReportingDemo.InsultsReport.rdlc,
Legal Matches are: [Model]